src/cargo/ops/cargo_rustc/compilation.rs

```diff
@@ -109,12 +109,17 @@ impl<'cfg> Compilation<'cfg> {
             }
         }
 
+	 let metadata = pkg.manifest().metadata().clone();
```

Comment:
metadata() does not return a  reference, so i think it needs to. Please correct me if am wrong. I just started on rust.